### PR TITLE
Allow gateway-api to function without Kubernetes RBAC

### PR DESCRIPTION
**Pull request by howardjohn**
This implements most of
https://docs.google.com/document/d/1gDhj058q2PKKI9SWpxE23ExEXdOkLiueT8JA_oEJ8Mo/edit#heading=h.bmkc9oooxqbj

Another PR will try to figure out the multicluster change
I tried to capture context in the comments in the code

### CODE REVIEW

**Comment by howardjohn** [comment]
Put a hold to add better tests

File: pilot/pkg/xds/sds.go [hunk 1]
```diff
@@ -102,50 +103,101 @@ func (s *SecretGen) proxyAuthorizedForSecret(proxy *model.Proxy, sr SecretResour
...
+	if err != nil {
+		log.Warnf("proxy %v is from an unknown cluster, cannot retrieve certificates: %v", proxy.ID, err)
+		return nil, model.DefaultXdsLogDetails, nil
+	}
+
+	resources := parseResources(w.ResourceNames, proxy)
+	authMode := getAuthorizationMode(resources, proxy)
+	if authMode != OnlyLocalCertificateReferences {
+		if err := secrets.Authorize(proxy.VerifiedIdentity.ServiceAccount, proxy.VerifiedIdentity.Namespace); err != nil {
+			log.Warnf("proxy %v is not authorized to receive secrets: %v", proxy.ID, err)
```
Comment (stevenctl) [inline pilot/pkg/xds/sds.go:185]: do we want to increment the metric here?

File: pilot/pkg/xds/sds.go [hunk 2]
```diff
@@ -102,50 +103,101 @@ func (s *SecretGen) proxyAuthorizedForSecret(proxy *model.Proxy, sr SecretResour
...
 	}
 	var updatedSecrets map[model.ConfigKey]struct{}
 	if !req.Full {
 		updatedSecrets = model.ConfigsOfKind(req.ConfigsUpdated, gvk.Secret)
 	}
+
+	// TODO: For the new gateway-api, we should always search the config namespace and stop reading across all clusters
+	secrets, err := s.secrets.ForCluster(proxy.Metadata.ClusterID)
+	if err != nil {
+		log.Warnf("proxy %v is from an unknown cluster, cannot retrieve certificates: %v", proxy.ID, err)
```
Comment (stevenctl) [inline pilot/pkg/xds/sds.go:177]: ditto: metric

File: pilot/pkg/xds/sds.go [hunk 3]
```diff
@@ -102,50 +103,101 @@ func (s *SecretGen) proxyAuthorizedForSecret(proxy *model.Proxy, sr SecretResour
...
 
+	for _, sr := range resources {
 		if updatedSecrets != nil {
 			if !containsAny(updatedSecrets, relatedConfigs(model.ConfigKey{Kind: gvk.Secret, Name: sr.Name, Namespace: sr.Namespace})) {
 				// This is an incremental update, filter out secrets that are not updated.
 				continue
 			}
 		}
 
-		if err := s.proxyAuthorizedForSecret(proxy, sr); err != nil {
```
Comment (stevenctl) [inline pilot/pkg/xds/sds.go:144]: Shouldn't we still do this for the `CrossNamespaceCertificateReferences` case?

Comment (howardjohn) [reply pilot/pkg/xds/sds.go:144]: We should - and we still do (we have tests as well). This however is dead code, so I removed the function. The new check is where we have `requested secret %v not accessible for proxy %v: SDS is currently only supporting accessing secret within the same namespace. `.

Would be good to have a sanity check that actually makes sense - coming back to this 2 weeks later I think it is but this is critical code

**Review by stevenctl — APPROVED** [review]

**Review by costinm — APPROVED** [review]
Seems safe...

**Pull request merged**